### ISSUE TITLE
Process is exiting  with exit code 139 (interrupted by signal 11: SIGSEGV) if not passing logger param to the Trainer

### ISSUE THREAD

**Issue by snarb** [PyTorchLightning/pytorch-lightning#8026]
Process is exiting  with exit code 139 (interrupted by signal 11: SIGSEGV) if not passing logger param to the Trainer

This code working fine:
```
myModule = myLightningModule()
trainer = pl.Trainer(max_epochs=FEATURE_EXTRACTION_EPOCHS, gpus=GPU_COUNT, accelerator="dp", precision=16, logger=None)
trainer.validate(myModule, val_dataloaders = myModule.valDl)
```

This one crashing a process:

    myModule = myLightningModule()
    trainer = pl.Trainer(max_epochs=FEATURE_EXTRACTION_EPOCHS, gpus=GPU_COUNT, accelerator="dp", precision=16)
    trainer.validate(myModule, val_dataloaders = myModule.valDl)

torch.__version__
1.8.0
pytorch_lightning.__version__
1.3.4

**Comment by kaushikb11** [comment]
@snarb Could you kindly create a minimal script/notebook to reproduce this issue?

**Comment by tchaton** [comment]
Dear @snarb,

It seems the logger is the source of failure in examples right ? Mind creating a reproducible script with the BoringModel ?

Best,
T.C

**Comment by snarb** [comment]
Found minal steps for reproducing:

    import torchvision # This torchvision import causes crash. If we are removing this import - no crash
    import pytorch_lightning as pl
    from torch.nn import functional as F
    from torch import nn
    from pytorch_lightning.core.lightning import LightningModule
    
    
    class LitMNIST(LightningModule):
    
      def __init__(self):
        super().__init__()
    
        # mnist images are (1, 28, 28) (channels, width, height)
        self.layer_1 = nn.Linear(28 * 28, 128)
        self.layer_2 = nn.Linear(128, 256)
        self.layer_3 = nn.Linear(256, 10)
    
      def forward(self, x):
        batch_size, channels, width, height = x.size()
    
        # (b, 1, 28, 28) -> (b, 1*28*28)
        x = x.view(batch_size, -1)
        x = self.layer_1(x)
        x = F.relu(x)
        x = self.layer_2(x)
        x = F.relu(x)
        x = self.layer_3(x)
    
        x = F.log_softmax(x, dim=1)
        return x
    
    
    def main():
        postersModel = LitMNIST()
        trainer = pl.Trainer()
        trainer.validate(postersModel)
        # trainer = pl.Trainer(logger=None)                  - call with logger=None will not cause crash
        print('Done')
    
    if __name__ == "__main__":
        main()


So looks like torchvision import together with logger param somehow causes the crash. 
torchvision.__version__
0.9.0
